src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
```

Comment:
I would be better to declare constructor which sets INITIAL_CAPACITY to the array while object creation
     private static final int INITIAL_CAPACITY = 10;
     private int size = 0;
     private Object elementData[] = {};

       public ArrayListCustom() {
           elementData = new Object[INITIAL_CAPACITY];
      }